label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.label.application.rule.config.service.v1.CreateLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.CreateLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.DeleteLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.DeleteLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRulesRequest;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRulesResponse;
+import org.hypertrace.label.application.rule.config.service.v1.LabelApplicationRule;
+import org.hypertrace.label.application.rule.config.service.v1.LabelApplicationRuleConfigServiceGrpc;
+import org.hypertrace.label.application.rule.config.service.v1.UpdateLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.UpdateLabelApplicationRuleResponse;
+
+public class LabelApplicationRuleConfigServiceImpl
+    extends LabelApplicationRuleConfigServiceGrpc.LabelApplicationRuleConfigServiceImplBase {
+  public static final String LABEL_APPLICATION_RULE_CONFIG_RESOURCE_NAME = "label-config";
+  public static final String LABEL_APPLICATION_RULE_CONFIG_RESOURCE_NAMESPACE =
+      "label-application-rules";
+  private final IdentifiedObjectStore<LabelApplicationRule> labelApplicationRuleStore;
+  private final LabelApplicationRuleValidator requestValidator;
+
+  public LabelApplicationRuleConfigServiceImpl(Channel configChannel) {
+    ConfigServiceBlockingStub configServiceBlockingStub =
+        ConfigServiceGrpc.newBlockingStub(configChannel)
+            .withCallCredentials(
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+    this.labelApplicationRuleStore =
+        new LabelApplicationRuleStore(
+            configServiceBlockingStub,
+            LABEL_APPLICATION_RULE_CONFIG_RESOURCE_NAMESPACE,
+            LABEL_APPLICATION_RULE_CONFIG_RESOURCE_NAME);
+    this.requestValidator = new LabelApplicationRuleValidatorImpl();
+  }
+
+  @Override
+  public void createLabelApplicationRule(
+      CreateLabelApplicationRuleRequest request,
+      StreamObserver<CreateLabelApplicationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      this.requestValidator.validateOrThrow(requestContext, request);
+      LabelApplicationRule labelApplicationRule =
+          LabelApplicationRule.newBuilder()
+              .setId(UUID.randomUUID().toString())
```

Comment:
nit/suggestion: Use `UuidGenerator` (which makes it easier to test, since that can be a mocked dependency)